pkg/build/generator/generator.go

```diff
@@ -513,6 +513,11 @@ func (g *BuildGenerator) generateBuildFromConfig(ctx kapi.Context, bc *buildapi.
 				Kind: "DockerImage",
 				Name: runtimeImageName,
 			}
+			// We're using single secret for pulling builder and runtime images.
+			// If builder image doesn't need a secret, let's check whether runtime image needs it or not.
+			if build.Spec.Strategy.SourceStrategy.PullSecret == nil {
+				build.Spec.Strategy.SourceStrategy.PullSecret = g.resolveImageSecret(ctx, builderSecrets, build.Spec.Strategy.SourceStrategy.RuntimeImage, bc.Namespace)
+			}
```

Comment:
>1) creds[builderPull,runtimePull]

@bparees How do you see this?

Currently [`resolveImageSecret()` creates keyring based on available secrets](https://github.com/openshift/origin/blob/6c50f2007a5426aa71a4aaa6e76302e58196c61e/pkg/build/generator/generator.go#L646-L662) and invokes `keyring.Lookup()` on it.

I can suppose that we should invoke `keyring.Lookup()` twice for each image. Is this what you suggested? Or we should modify `keyring.Lookup()` method for accepting two image specs?